### PR TITLE
AAE-19797 Make user type parameterized in search query

### PR DESCRIPTION
**Pull request by tom-dal**
Issue link: AAE-19797

### CODE REVIEW

**Review by mavotto — APPROVED** [review]

File: activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/identity/UserSearchParams.java [hunk 1]
```diff
@@ -22,6 +22,7 @@ public class UserSearchParams {
     private String search;
     private Set<String> groups;
     private Set<String> roles;
+    private Set<String> types;
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/identity/UserSearchParams.java]: could u user a enum?

**Review by jesty — CHANGES_REQUESTED** [review]

File: activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/identity/UserSearchParams.java (hunk 1 above)
Comment (tom-dal) [reply activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/identity/UserSearchParams.java]: Done. I also added an InvalidUserTypeException which will be convenient in indentity-adapter-service.

File: activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/identity/model/UserType.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.identity.model;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum UserType {
+    INTERACTIVE,
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/identity/model/UserType.java]: what about ALL type?

**Review by jesty — CHANGES_REQUESTED** [review]

File: activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java [hunk 3]
```diff
@@ -74,6 +80,13 @@ public List<User> findUsers(UserSearchParams userSearchParams) {
         }
     }
 
+    private List<User> searchUsers(String searchKey, UserTypeSearchParam userType) {
+        return switch (userType) {
+            case INTERACTIVE -> searchUsers(searchKey);
+            case ALL -> searchUsersByUsername(searchKey);
```
Comment (jesty) [inline activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/KeycloakManagementService.java:88]: can you add a comment for explaning why you have to do this switch? thanks

**Review by jesty — CHANGES_REQUESTED** [review]

**Review by jesty — APPROVED** [review]
LGTM

**Pull request merged**